### ISSUE TITLE
fmtstr: raise error when the resulting string contains '\n'

### ISSUE THREAD

**Issue by ishitatsuyuki** [Gallopsled/pwntools#1073]
Apart from `\0` which is an issue of the current implementation, `\n` is often a special delimiter that result in a faulty payload if the target is using `fgets()` and the address contains `0a`.

We should raise an exception for manual hack in such case, and also provide an opt-out option for using with non-line based cases. The parameter suggestion would be `bad_chars=['\n']`.

**Comment by zachriggle** [comment]
This is work that would belong in #827.  I don't intend to make any changes to the current `fmtstr` module, since it has a great many limitations and design issues.

The most recent status is in my comment here: https://github.com/Gallopsled/pwntools/pull/827#issuecomment-350085470

tl;dr I don't have any time to dedicate to this currently, but I've put a lot of work into making a fully-automated format string exploitation tool that works on all architecture.
